src/apps/command_view_app.tsx

```diff
@@ -78,9 +79,12 @@ class CommandViewApp extends Component<MyProps, MyState> {
     this.uischema = build.form;
     this.initialData = build.model;
     let requestData = null;
-    if (this.props.location.state) {
+    const pourItAgainRequest = CacheService.popQueue(
```

Comment:
I think this push / pop thing you're doing is fine and we should just go with it, but I'll point out just for awareness that you've altered the race condition rather than removed it with this solution.  So now the scenario is that if you middle click "pour it again" from two different pages, there's a race between the two and whichever one loads first will get the last thing you pushed onto the queue.  So it's possible that the new command pages that open each get the other's data loaded (or attempted to load.  The two commands may not have compatible data to put in the forms).

This feels highly unlikely to occur in practice.  I mention it for your own awareness just so that you can have these sorts of race conditions in mind for future cases where it might be of more importance.